nodes/ati_ft_sensor_wrench_pub.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Title: ati_ft_sensor_wrench_pub.cpp
+ *
+ * Description: This file is part of ati_ft_sensor package. It publishes the
+ * wrench data from the ATI force torque sensor. It also publishes the wrench
+ * data to the faked_forces_controller based on a passed parameter, serving as a
+ * generic proxy for admittance control using PickNik's admittance controller.
+ * Follow the general template to replicate this for another FT sensor.
+ *
+ * Author: Emmanuel Akita
+ * Email:  efakita@utexas.edu
+ * Created: 4 Oct, 2023
+ *
+ */
+
+#include <AtiFTSensor.h>
+#include <chrono>
+
+// ROS includes
+#include "geometry_msgs/msg/wrench_stamped.hpp"
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/float64_multi_array.hpp"
+
+class ATIWrenchPublisher : public rclcpp::Node {
+  // Constructor
+ public:
+  ATIWrenchPublisher() : Node("ati_wrench_publisher") {
+    initializeNode();
+
+    // Create a timer that calls the publish function every millisecond.
+    timer_ = this->create_wall_timer(
+        std::chrono::milliseconds(1),
+        std::bind(&ATIWrenchPublisher::publishSensorData, this));
+
+    if (!sensor_.initialize(ip_address_)) {
+      RCLCPP_ERROR(this->get_logger(), "Could not initialize the sensor.");
+    }
+  }
+
+  // Destructor
+  ~ATIWrenchPublisher() {
+    // Stop the timer.
+    timer_->cancel();
+  }
+
+ private:
+  rclcpp::TimerBase::SharedPtr timer_;
+  ati_ft_sensor::AtiFTSensor sensor_;
+  rclcpp::Publisher<geometry_msgs::msg::WrenchStamped>::SharedPtr pub_wrench_;
+  rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr
+      pub_fake_controller_wrench_;
+  std::string frame_id_, ip_address_;
+  bool fake_controller_wrench_;
+
+  void initializeNode() {
+    // Get the required parameters
+    this->declare_parameter("frame_id", rclcpp::PARAMETER_STRING);
+    this->declare_parameter("ip_address", rclcpp::PARAMETER_STRING);
+    this->declare_parameter("pub_faked_wrench", rclcpp::PARAMETER_BOOL);
+
+    rclcpp::Parameter frame_id = this->get_parameter("frame_id");
+    rclcpp::Parameter pub_faked_wrench =
+        this->get_parameter("pub_faked_wrench");
+    rclcpp::Parameter ip_address = this->get_parameter("ip_address");
+
+    frame_id_ = frame_id.as_string();
+    ip_address_ = ip_address.as_string();
+    fake_controller_wrench_ = pub_faked_wrench.as_bool();
+
+    // Create a publisher for the sensor data topic.
+    pub_wrench_ = this->create_publisher<geometry_msgs::msg::WrenchStamped>(
+        "wrench_data", rclcpp::SensorDataQoS());
+
+    if (fake_controller_wrench_)
+      pub_fake_controller_wrench_ =
+          this->create_publisher<std_msgs::msg::Float64MultiArray>(
```

Comment:
I don't fully understand why you need a different type of msg to publish the fake control wrench...  Why don't you use the same wrenchStamped or even just wrench

Reply:
That's the msg type that the fake_controller accepts